src/script_parser.rs

```diff
@@ -0,0 +1,178 @@
+    let invalid_name = re.is_match(&name);
+    name == &name.to_uppercase() && name.len() > 1 && name.len() < 30 && !invalid_name
+}
+
+/// Identifies the characters and gets their line counts and speaks counts
+pub fn parse_characters(line_chunks: Vec<String>) -> HashMap<String, Character> {
+    let mut characters: HashMap<String, Character> = HashMap::new();
+
+    // loop through the script and create/add to characters
+    for chunk in line_chunks {
+        let name: String;
+        // Grab the name
+        let potential_name_line: Vec<&str> = chunk.split(".").collect();
+        // Check if the name has an honorific attached
+        let re = Regex::new(r"^MR$|^MRS$|^MS$|^MISS$|^DR$|^JR$|^SR$|^PROF$|^REV$|^ST$|^COL$|^GEN$|^GOV$|^PRES$|^LT$").unwrap();
+        if re.is_match(potential_name_line[0]) {
+            name = format!("{}{}", potential_name_line[0], potential_name_line[1]);
+        } else {
+            name = format!("{}", potential_name_line[0]);
+        }
+        // Check that the name is legit, if it isn't move to the next chunk
+        if !is_name(&name) {
+            continue;
+        }
+
+        let mut curr_char = characters.entry(name.clone()).or_insert(Character {
+            name: name.clone(),
+            line_count: 0,
+            speaks_count: 0,
+            id: None,
+            actor_id: None,
+        });
+        // Check whether lines start on the same line as the character name, or on the next line
+        let lines_in_chunk: Vec<&str> = chunk.split("\n").collect();
+        if lines_in_chunk.len() > name.len() {
+            curr_char.line_count += lines_in_chunk.len() as u32;
+        } else {
+            // Minus one to account for the first line of the chunk that is just the name
+            curr_char.line_count += lines_in_chunk.len() as u32 - 1;
+        }
+        curr_char.speaks_count += 1;
+    }
+
+    characters
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::characters::Character;
+    use std::collections::HashMap;
+
+    #[test]
+    fn test_parse_title() {
+        let sample = "AUTHOR: Steve\n\nTitle: Test 1\n\nACTOR 1.".to_string();
+        let title = parse_title(&sample).unwrap();
+        assert_eq!(title, "Test 1");
+
+        let sample = "AUTHOR: Steve\n\nTITLE: Test 1\n\nACTOR 1.".to_string();
+        let title = parse_title(&sample).unwrap_err();
+        assert_eq!(title, 452)
+    }
+
+    #[test]
+    fn test_parse_lines() {
+        let sample = "Title: Test 1\n\nACTOR 1.\nSome test lines of a play\nthen some more lines that shall say things.\n\nACTOR 2.\nWhat a silly thing to say.".to_string();
+        let line_chunks = parse_lines(&sample);
+        let expected = vec![
+            "Title: Test 1".to_string(),
+            "ACTOR 1.\nSome test lines of a play\nthen some more lines that shall say things."
+                .to_string(),
+            "ACTOR 2.\nWhat a silly thing to say.".to_string(),
```

Comment:
If all the tests assert against static strings that have to_string() called on them, it might be simpler to flip where the conversion happens.  
Could we have a vec of `string slices` and then convert the `String` output to a slice when we assert?